go.mod

```diff
@@ -4,7 +4,7 @@ go 1.16
 
 // CAVEAT: This introduces a circular dependency. If you change this line, you MUST change the "exclude"
 // directive at the bottom of the file as well.
-require github.com/stackrox/scanner v0.0.0-20220408151911-460993206ee4
+require github.com/stackrox/scanner v0.0.0-20220414151742-79b8a1fe00c0
```

Comment:
I'm trying to update this to the latest version of https://github.com/stackrox/scanner/pull/688, but I run into the following:
```
rtannenb:~/go/src/github.com/stackrox/stackrox$ go mod tidy
go: finding module for package go.opentelemetry.io/otel/unit
go: finding module for package go.opentelemetry.io/otel/semconv
go: finding module for package go.opentelemetry.io/otel/internal/metric
go: found go.opentelemetry.io/otel/internal/metric in go.opentelemetry.io/otel/internal/metric v0.27.0
go: finding module for package go.opentelemetry.io/otel/metric/registry
go: finding module for package go.opentelemetry.io/otel/semconv
github.com/stackrox/rox/pkg/auth/authproviders/saml imports
	github.com/cloudflare/cfssl/helpers imports
	github.com/google/certificate-transparency-go imports
	go.etcd.io/etcd/v3 imports
	go.etcd.io/etcd/tests/v3/integration imports
	go.etcd.io/etcd/server/v3/embed imports
	go.opentelemetry.io/otel/semconv: module go.opentelemetry.io/otel@latest found (v1.6.3), but does not contain package go.opentelemetry.io/otel/semconv
github.com/stackrox/rox/pkg/auth/authproviders/saml imports
	github.com/cloudflare/cfssl/helpers imports
	github.com/google/certificate-transparency-go imports
	go.etcd.io/etcd/v3 imports
	go.etcd.io/etcd/tests/v3/integration imports
	go.etcd.io/etcd/server/v3/embed imports
	go.opentelemetry.io/otel/exporters/otlp imports
	go.opentelemetry.io/otel/sdk/metric/controller/basic imports
	go.opentelemetry.io/otel/metric/registry: module go.opentelemetry.io/otel/metric@latest found (v0.29.0), but does not contain package go.opentelemetry.io/otel/metric/registry
```

I have been looking into this, but I am wondering if anyone has any ideas @connorgorman @viswajithiii ?

Reply:
I think updating the `exclude github.com/stackrox/rox` directive at the bottom will help?

Reply:
The version in scanner is still the same, though: https://github.com/stackrox/scanner/blob/master/go.mod#L51

Now scanner has the replace directive `github.com/stackrox/rox => github.com/stackrox/stackrox v0.0.0-20210914215712-04ef1c1cfc62` so I tried adding `exclude github.com/stackrox/stackrox v0.0.0-20210914215712-04ef1c1cfc62` but it didn't help

Reply:
Looked into it some more, and I found https://github.com/golang/go/wiki/Modules#gomod which says 

> `exclude` and `replace` directives only operate on the current (“main”) module. `exclude` and `replace` directives in modules other than the main module are ignored when building the main module. The `replace` and `exclude` statements, therefore, allow the main module complete control over its own build, without also being subject to complete control by dependencies. (See FAQ [below](https://github.com/golang/go/wiki/Modules#when-should-i-use-the-replace-directive) for a discussion of when to use a `replace` directive).

This tells me the `replace github.com/stackrox/rox => github.com/stackrox/stackrox v0.0.0-20210914215712-04ef1c1cfc62` in scanner should have no effect on this repo. Since Scanner still has `require github.com/stackrox/rox v0.0.0-20210914215712-9ac265932e28`, then all we'd need in this repo is `exclude github.com/stackrox/rox v0.0.0-20210914215712-9ac265932e28`, which we already have. However, I keep running into this error.

The following shows me this package should have nothing to do with scanner...?
```
rtannenb:~/go/src/github.com/stackrox/stackrox$ go mod why -m go.opentelemetry.io/otel
# go.opentelemetry.io/otel
github.com/stackrox/rox/pkg/auth/authproviders/saml
github.com/cloudflare/cfssl/helpers
github.com/google/certificate-transparency-go
go.etcd.io/etcd/v3
go.etcd.io/etcd/tests/v3/integration
go.etcd.io/etcd/server/v3/embed
go.opentelemetry.io/otel/propagation
rtannenb:~/go/src/github.com/stackrox/stackrox$ go mod why -m go.opentelemetry.io/otel/metric
# go.opentelemetry.io/otel/metric
github.com/stackrox/rox/pkg/auth/authproviders/saml
github.com/cloudflare/cfssl/helpers
github.com/google/certificate-transparency-go
go.etcd.io/etcd/v3
go.etcd.io/etcd/tests/v3/integration
go.etcd.io/etcd/server/v3/embed
go.opentelemetry.io/otel/exporters/otlp
go.opentelemetry.io/otel/metric
```

Reply:
It is due to the version of `github.com/containers/image/v5`. I will downgrade it https://github.com/stackrox/scanner/pull/706